### ISSUE TITLE
Uncaught PHP Exception when redirect to SAML SSO

### ISSUE THREAD

**Issue by cubesky** [Part-DB/Part-DB-server#318]
**Describe the bug**
When using SAML login method, redirect to SSO server is denied.

When I manully look at PartDB logs for the target redirect url, I copy it and paste to my browser (Keycloak url), login and redirect back, PartDB successfully logined. 

**To Reproduce**
1. Config the SAML following the Configuration document with keycloak server. (Keycloak on `https://sso.example.com/`)
2. Config PartDB using docker.
3. Press the SSO button.

**Expected behavior**
Redirect to the Keycloak login page.

**Screenshots**
![image](https://github.com/Part-DB/Part-DB-server/assets/8038511/c21442fe-b49a-45e0-b37e-4280fdc64181)

**Server Side**
- Part-DB Version: Part-DB 1.5.2
- PHP Version: PHP 8.1.20
- Database Server: Sqlite

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 114.0.5735.199

**Smartphone (please complete the following information):**
 - Device: Not Use
 - OS: Not Use
 - Browser Not Use
 - Version Not Use

**Additional context**
The following log replaced my domain with `example.com`

```
partdb    | [10-Jul-2023 20:10:16] NOTICE: fpm is running, pid 24
partdb    | [10-Jul-2023 20:10:16] NOTICE: ready to handle connections
partdb    | [10-Jul-2023 20:10:16] NOTICE: systemd monitor interval set to 10000ms
partdb    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 192.168.64.2. Set the 'ServerName' directive globally to suppress this message
partdb    | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 192.168.64.2. Set the 'ServerName' directive globally to suppress this message
partdb    | [Mon Jul 10 20:10:16.737909 2023] [mpm_event:notice] [pid 1:tid 140665374002496] AH00489: Apache/2.4.56 (Debian) configured -- resuming normal operations
partdb    | [Mon Jul 10 20:10:16.738109 2023] [core:notice] [pid 1:tid 140665374002496] AH00094: Command line: 'apache2 -D FOREGROUND'
partdb    | - -  10/Jul/2023:20:10:21 +0000 "GET /index.php" 302
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:21 +0000] "GET / HTTP/1.1" 302 875 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:24 +0000 "GET /index.php" 200
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:24 +0000] "GET /en/ HTTP/1.1" 200 6249 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:25 +0000 "POST /index.php" 204
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:25 +0000] "POST /csp/report HTTP/1.1" 204 748 "https://partdb.example.com/en/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:25 +0000 "GET /index.php" 200
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:25 +0000] "GET /en/tree/devices HTTP/1.1" 200 937 "https://partdb.example.com/en/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:25 +0000 "GET /index.php" 200
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:25 +0000] "GET /en/tree/tools HTTP/1.1" 200 2943 "https://partdb.example.com/en/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:25 +0000 "GET /index.php" 200
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:25 +0000] "GET /en/tree/categories HTTP/1.1" 200 936 "https://partdb.example.com/en/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:27 +0000 "GET /index.php" 200
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:27 +0000] "GET /en/login?_target_path=/ HTTP/1.1" 200 6226 "https://partdb.example.com/en/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:27 +0000 "POST /index.php" 204
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:27 +0000] "POST /csp/report HTTP/1.1" 204 748 "https://example.com/en/login?_target_path=/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | {"message":"Matched route \"saml_login\".","context":{"route":"saml_login","route_parameters":{"_route":"saml_login","_controller":"Nbgrp\\OneloginSamlBundle\\Controller\\Login","idp":null},"request_uri":"https://partdb.example.com/saml/login","method":"GET"},"level":200,"level_name":"INFO","channel":"request","datetime":"2023-07-10T20:10:28.250504+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Checking for authenticator support.","context":{"firewall_name":"main","authenticators":4},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251811+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Checking support on authenticator.","context":{"firewall_name":"main","authenticator":"Scheb\\TwoFactorBundle\\Security\\Http\\Authenticator\\TwoFactorAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251834+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Authenticator does not support the request.","context":{"firewall_name":"main","authenticator":"Scheb\\TwoFactorBundle\\Security\\Http\\Authenticator\\TwoFactorAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251855+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Checking support on authenticator.","context":{"firewall_name":"main","authenticator":"Nbgrp\\OneloginSamlBundle\\Security\\Http\\Authenticator\\SamlAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251868+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Authenticator does not support the request.","context":{"firewall_name":"main","authenticator":"Nbgrp\\OneloginSamlBundle\\Security\\Http\\Authenticator\\SamlAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251882+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Checking support on authenticator.","context":{"firewall_name":"main","authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\FormLoginAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251895+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Authenticator does not support the request.","context":{"firewall_name":"main","authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\FormLoginAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251909+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Checking support on authenticator.","context":{"firewall_name":"main","authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\RememberMeAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251922+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Authenticator does not support the request.","context":{"firewall_name":"main","authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\RememberMeAuthenticator"},"level":100,"level_name":"DEBUG","channel":"security","datetime":"2023-07-10T20:10:28.251938+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"User Deprecated: Using XML mapping driver with XSD validation disabled is deprecated and will not be supported in Doctrine ORM 3.0. (XmlDriver.php:65 called by App_KernelDockerContainer.php:290, https://github.com/doctrine/orm/pull/6728, package doctrine/orm)","context":{"exception":{"class":"ErrorException","message":"User Deprecated: Using XML mapping driver with XSD validation disabled is deprecated and will not be supported in Doctrine ORM 3.0. (XmlDriver.php:65 called by App_KernelDockerContainer.php:290, https://github.com/doctrine/orm/pull/6728, package doctrine/orm)","code":0,"file":"/var/www/html/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:209","trace":["/var/www/html/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:107","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php:65","/var/www/html/var/cache/docker/ContainerExkoa7x/App_KernelDockerContainer.php:290","/var/www/html/var/cache/docker/ContainerExkoa7x/App_
partdb    | KernelDockerContainer.php:283","/var/www/html/vendor/symfony/var-exporter/Internal/LazyObjectState.php:97","/var/www/html/vendor/symfony/var-exporter/LazyGhostTrait.php:176","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:825","/var/www/html/src/EventSubscriber/UserSystem/UpgradePermissionsSchemaSubscriber.php:54","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:260","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:220","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:56","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:142","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:74","/var/www/html/vendor/symfony/http-kernel/Kernel.php:197","/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35","/var/www/html/vendor/autoload_runtime.php:29","/var/www/html/public/index.php:5"]}},"level":200,"level_name":"INFO","channel":"deprecation","datetime":"2023-07-10T20:10:28.252697+00:00","extra":{"token":null,"u
partdb    | rl":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"User Deprecated: Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead","context":{"exception":{"class":"ErrorException","message":"User Deprecated: Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead","code":0,"file":"/var/www/html/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php:192","trace":["/var/www/html/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php:101","/var/www/html/vendor/doctrine/dbal/src/Connection.php:384","/var/www/html/vendor/doctrine/dbal/src/Connection.php:450","/var/www/html/vendor/doctrine/dbal/src/Connection.php:408","/var/www/html/vendor/doctrine/dbal/src/Connection.php:316","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:206","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:3286","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:239","/var
partdb    | /www/html/src/Repository/UserRepository.php:50","/var/www/html/src/EventSubscriber/UserSystem/UpgradePermissionsSchemaSubscriber.php:54","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:260","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:220","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:56","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:142","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:74","/var/www/html/vendor/symfony/http-kernel/Kernel.php:197","/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35","/var/www/html/vendor/autoload_runtime.php:29","/var/www/html/public/index.php:5"]}},"level":200,"level_name":"INFO","channel":"deprecation","datetime":"2023-07-10T20:10:28.254654+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"User Deprecated: Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead","context":{"exception":{"class":"ErrorException","message":"User Deprecated: Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead","code":0,"file":"/var/www/html/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php:192","trace":["/var/www/html/vendor/symfony/doctrine-bridge/ContainerAwareEventManager.php:101","/var/www/html/vendor/doctrine/dbal/src/Connection.php:384","/var/www/html/vendor/doctrine/dbal/src/Connection.php:450","/var/www/html/vendor/doctrine/dbal/src/Connection.php:408","/var/www/html/vendor/doctrine/dbal/src/Connection.php:316","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php:206","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:3286","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:239","/var
partdb    | /www/html/src/Repository/UserRepository.php:50","/var/www/html/src/EventSubscriber/UserSystem/UpgradePermissionsSchemaSubscriber.php:54","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:260","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:220","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:56","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:142","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:74","/var/www/html/vendor/symfony/http-kernel/Kernel.php:197","/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35","/var/www/html/vendor/autoload_runtime.php:29","/var/www/html/public/index.php:5"]}},"level":200,"level_name":"INFO","channel":"deprecation","datetime":"2023-07-10T20:10:28.254988+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"User Deprecated: Subscribing to postConnect events is deprecated. Implement a middleware instead. (Connection.php:389 called by Connection.php:450, https://github.com/doctrine/dbal/issues/5784, package doctrine/dbal)","context":{"exception":{"class":"ErrorException","message":"User Deprecated: Subscribing to postConnect events is deprecated. Implement a middleware instead. (Connection.php:389 called by Connection.php:450, https://github.com/doctrine/dbal/issues/5784, package doctrine/dbal)","code":0,"file":"/var/www/html/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:209","trace":["/var/www/html/vendor/doctrine/deprecations/lib/Doctrine/Deprecations/Deprecation.php:107","/var/www/html/vendor/doctrine/dbal/src/Connection.php:389","/var/www/html/vendor/doctrine/dbal/src/Connection.php:450","/var/www/html/vendor/doctrine/dbal/src/Connection.php:408","/var/www/html/vendor/doctrine/dbal/src/Connection.php:316","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entit
partdb    | y/BasicEntityPersister.php:206","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:3286","/var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:239","/var/www/html/src/Repository/UserRepository.php:50","/var/www/html/src/EventSubscriber/UserSystem/UpgradePermissionsSchemaSubscriber.php:54","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:260","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:220","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:56","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:142","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:74","/var/www/html/vendor/symfony/http-kernel/Kernel.php:197","/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35","/var/www/html/vendor/autoload_runtime.php:29","/var/www/html/public/index.php:5"]}},"level":200,"level_name":"INFO","channel":"deprecation","datetime":"2023-07-10T20:10:28.255220+00:00","extra":{"token":null,"url":"/saml/lo
partdb    | gin","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"External redirect detected from https://partdb.example.com/saml/login to https://sso.example.com/auth/realms/sso/protocol/saml?SAMLRequest=(...This a vaild sso url)
partdb    | ujopXtTRrLmdHVWKd4kFYIcQ9e7a3yHo7b0LdBHrMrK56dIqgsVfNLOniNRm9vMtW5qpCna707RLwi8B3I4vw%2FPp5iFAtD6YA%2FVECgYIF5AOdfjj2pqmNt%2BVXtlkCkGj6SAeOTP5%2B%2Bq6AaoEEOPZPjkrqrAI4mjFXCscfpuxMMxpoJ7rnXGobFkqdICy5l10oW6PUGCDcfkfxNpilKBzKq1uhopuRvkwyJXvoxAkZWICEA%3D%3D","context":{},"level":300,"level_name":"WARNING","channel":"security","datetime":"2023-07-10T20:10:28.269460+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | {"message":"Uncaught PHP Exception Symfony\\Component\\HttpKernel\\Exception\\HttpException: \"Invalid Redirect Given: https://sso.example.com/auth/realms/sso/protocol/saml?SAMLRequest=(...This a vaild sso url)
partdb    | 911biiBD9W279vwiG%2BxjAsXAqvjoujopXtTRrLmdHVWKd4kFYIcQ9e7a3yHo7b0LdBHrMrK56dIqgsVfNLOniNRm9vMtW5qpCna707RLwi8B3I4vw%2FPp5iFAtD6YA%2FVECgYIF5AOdfjj2pqmNt%2BVXtlkCkGj6SAeOTP5%2B%2Bq6AaoEEOPZPjkrqrAI4mjFXCscfpuxMMxpoJ7rnXGobFkqdICy5l10oW6PUGCDcfkfxNpilKBzKq1uhopuRvkwyJXvoxAkZWICEA%3D%3D\" at /var/www/html/vendor/nelmio/security-bundle/src/EventListener/ExternalRedirectListener.php line 100","context":{"exception":{"class":"Symfony\\Component\\HttpKernel\\Exception\\HttpException","message":"Invalid Redirect Given: https://sso.example.com/auth/realms/sso/protocol/saml?SAMLRequest=(...This a vaild sso url)
partdb    | j50uLNemdsKej%2FQUI%2FbBGVDw5tKa81ViDVQUHu%2FZKZFYV5nrvmEUw9sjV4ffbb4eepg7ydQZec4ET9uSmda4XN1cBJSDrnuma7LZ8XbqC2cEjuzpzksqlzcOpez8bmza2BdGdnVjjzxtK5Sf8U71yzO7aT3mX79odKXgE%3D&RelayState=http%3A%2F%2Fpartdb.example.com%2Fsaml%2Flogin&SigAlg=http%3A%2F%2Fwww.w3.org%2F2001%2F04%2Fxmldsig-more%23rsa-sha256&Signature=SVjMNfBQyZKsYgXthWiEPAwyQl5ECpxPW%2BSz8t3prdoWEdDb6F74aorQSTGD6LJuYhnGQq8Fl%2BBE911biiBD9W279vwiG%2BxjAsXAqvjoujopXtTRrLmdHVWKd4kFYIcQ9e7a3yHo7b0LdBHrMrK56dIqgsVfNLOniNRm9vMtW5qpCna707RLwi8B3I4vw%2FPp5iFAtD6YA%2FVECgYIF5AOdfjj2pqmNt%2BVXtlkCkGj6SAeOTP5%2B%2Bq6AaoEEOPZPjkrqrAI4mjFXCscfpuxMMxpoJ7rnXGobFkqdICy5l10oW6PUGCDcfkfxNpilKBzKq1uhopuRvkwyJXvoxAkZWICEA%3D%3D","code":0,"file":"/var/www/html/vendor/nelmio/security-bundle/src/EventListener/ExternalRedirectListener.php:100","trace":["/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:260","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:220","/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php:56","/
partdb    | var/www/html/vendor/symfony/http-kernel/HttpKernel.php:199","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:187","/var/www/html/vendor/symfony/http-kernel/HttpKernel.php:74","/var/www/html/vendor/symfony/http-kernel/Kernel.php:197","/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35","/var/www/html/vendor/autoload_runtime.php:29","/var/www/html/public/index.php:5"]}},"level":400,"level_name":"ERROR","channel":"request","datetime":"2023-07-10T20:10:28.270540+00:00","extra":{"token":null,"url":"/saml/login","ip":"10.42.10.4","http_method":"GET","server":"partdb.example.com","referrer":"https://partdb.example.com/en/login?_target_path=/"}}
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:28 +0000] "GET /saml/login HTTP/1.1" 403 4286 "https://partdb.example.com/en/login?_target_path=/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
partdb    | - -  10/Jul/2023:20:10:28 +0000 "GET /index.php" 403
partdb    | - -  10/Jul/2023:20:10:28 +0000 "POST /index.php" 204
partdb    | 10.42.1.2 - - [10/Jul/2023:20:10:28 +0000] "POST /csp/report HTTP/1.1" 204 748 "https://partdb.example.com/saml/login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
```

docker-compose.yml
```yaml
version: '3.3'
services:
  partdb:
    container_name: partdb
    # By default Part-DB will be running under Port 8080, you can change it here
    ports:
      - '18580:80'
    volumes:
      # By default
      - ./uploads:/var/www/html/uploads
      - ./public_media:/var/www/html/public/media
      - ./db:/var/www/html/var/db
    restart: unless-stopped
    image: jbtronics/part-db1:latest
    environment:
      # Put SQLite database in our mapped folder. You can configure some other kind of database here too.
      - DATABASE_URL=sqlite:///%kernel.project_dir%/var/db/app.db
      # In docker env logs will be redirected to stderr
      - APP_ENV=docker

      # You can configure Part-DB using environment variables
      # Below you can find the most essential ones predefined
      # However you can add add any other environment configuration you want here
      # See .env file for all available options or https://docs.part-db.de/configuration.html

      # The language to use serverwide as default (en, de, ru, etc.)
      - DEFAULT_LANG=en
      # The default timezone to use serverwide (e.g. Europe/Berlin)
      - DEFAULT_TIMEZONE=Asia/Shanghai
      # The currency that is used inside the DB (and is assumed when no currency is set). This can not be changed later, so be sure to set it the currency used in your country
      - BASE_CURRENCY=CNY
      # The name of this installation. This will be shown as title in the browser and in the header of the website
      - INSTANCE_NAME=ExampleName

      # Allow users to download attachments to the server by providing an URL
      # This could be a potential security issue, as the user can retrieve any file the server has access to (via internet)
      - ALLOW_ATTACHMENT_DOWNLOADS=0
      # Use gravatars for user avatars, when user has no own avatar defined
      - USE_GRAVATAR=0

      # Override value if you want to show to show a given text on homepage.
      # When this is empty the content of config/banner.md is used as banner
      - BANNER=LiYin Parts
      - SAMLP_SP_PRIVATE_KEY=MIIE(...VaildCert...)=
      - SAML_SP_X509_CERT=MIIC(...VaildCert...)=
      - SAML_SP_ENTITY_ID=partdb
      - SAML_IDP_ENTITY_ID=https://sso.example.com/auth/realms/sso
      - SAML_IDP_SINGLE_SIGN_ON_SERVICE=https://sso.example.com/auth/realms/sso/protocol/saml
      - SAML_IDP_SINGLE_LOGOUT_SERVICE=https://sso.example.com/auth/realms/sso/protocol/saml
      - SAML_IDP_X509_CERT=MIIC(...VaildCert...)=
      - DEFAULT_URI=https://partdb.example.com/
      - SAML_ENABLED=1
      - TRUSTED_PROXIES=10.42.1.2,192.168.208.1
      - TRUSTED_HOSTS=partdb.example.com
```

Nginx
```
server {
    listen 10.42.1.2:443 ssl http2;
    ssl_certificate /etc/nginx/ssl/example/cert.pem;
    ssl_certificate_key /etc/nginx/ssl/example/key.pem;
    ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
    server_name partdb.example.com;
    client_max_body_size 1024M;
    include /etc/nginx/default.d/*.conf;
    gzip on;
    gzip_types text/plain application/javascript application/x-javascript text/css application/xml text/javascript application/x-httpd-php image/jpeg image/gif image/png application/vnd.ms-fontobject font/ttf font/opentype font/x-woff image/svg+xml;
    gzip_buffers 32 4k;
    gzip_disable "MSIE [1-6]\.";
    location / {
        if ($ssl_protocol != "") {
            add_header 'Content-Security-Policy' 'Upgrade-Insecure-Requests';
        }
        proxy_http_version 1.1;
        proxy_redirect off;
        proxy_set_header REMOTE-HOST $remote_addr;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Scheme $scheme;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_pass http://127.0.0.1:18580;
    }
}
```

Keycloak

![image](https://github.com/Part-DB/Part-DB-server/assets/8038511/86a53e09-a9d1-4f81-8762-f74913623657)

**Comment by jbtronics** [comment]
Interesting. That is not really an exception caused by some erroneous code, but is a basically a security feature, which should prevent url redirection attack, as a last security net. However, as SAML requires an external redirect, it prevents SAML in default configuration. (In my dev environment, keycloak and Part-DB are both on localhost, so I never noticed it before). 

I will try to find a solution, that the domain is automatically whitelisted. But for now you can go to `config/packages/nelmio_security.yaml.` andf find the line with `external_redirects`. Change the section to the following:

```yaml
    external_redirects:
        abort: true
        log: true
        allow_list:
             - your.idp.domain.com
```

Or you can disable the external redirect blocking completly by setting the `abort` key to `false`.
Afterward you have to run `php bin/console cache:clear` and the redirect should work as intended.

**Comment by cubesky** [comment]
Ok! It works!

**Comment by jbtronics** [comment]
This will be fixed in version 1.6.0
